### PR TITLE
NOISSUE - Add VM state machine and filter on qemu logs

### PR DESCRIPTION
**Pull request by SammyOina**
# What type of PR is this?

## What does this do?
- **New Features**
	- Enhanced state management for virtual machines, allowing for dynamic event reporting based on actual VM states.
	- Introduced a state machine for managing VM lifecycle events, improving clarity in state transitions during operations.
	- Added new logging functionalities that utilize state information for better event tracking.
	- New methods in the VM interface and mock for retrieving and transitioning VM states.

- **Bug Fixes**
	- Corrected event type reporting for broken connections to reflect the actual VM state.

- **Refactor**
	- Centralized log message handling to improve modularity and reduce code duplication.

### CODE REVIEW

**Review by WashingtonKK — CHANGES_REQUESTED** [review]
image

I have a duplicate log here

**Comment by SammyOina** [comment]
> image
> 
> I have a duplicate log here

Unrelated to this pr, will be solved on prism side

File: manager/service.go [hunk 1]
```diff
@@ -185,7 +185,10 @@ func (ms *managerService) Run(ctx context.Context, c *manager.ComputationRunReq)
 		return "", err
 	}
 
-	ms.vms[c.Id].Transition(manager.VmRunning)
+	if err := ms.vms[c.Id].Transition(manager.VmRunning); err != nil {
+		ms.logger.Warn("Failed to transition VM state", "computation", c.Id, "error", err)
```
Comment (WashingtonKK) [inline manager/service.go:189]: Use the  typed string for "error" here, or a constant

File: manager/service.go [hunk 2]
```diff
@@ -326,7 +329,10 @@ func (ms *managerService) restoreVMs() error {
 			continue
 		}
 
-		cvm.Transition(manager.VmRunning)
+		if err := cvm.Transition(manager.VmRunning); err != nil {
+			ms.logger.Warn("Failed to transition VM state", "computation", state.ID, "error", err)
```
Comment (WashingtonKK) [inline manager/service.go:333]: Same here

**Review by WashingtonKK — CHANGES_REQUESTED** [review]

File: manager/service.go (hunk 1 above)
Comment (SammyOina) [reply manager/service.go:189]: it is a log message we don't need a constant for that here

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**